src/com/google/javascript/jscomp/Compiler.java

```diff
@@ -1677,6 +1683,34 @@ private void repartitionInputs() {
     rebuildInputsFromModules();
   }
 
+  /**
+   * Transforms JSON files to a module export that closure compiler can
+   * process and keeps track of any "main" entries in package.json files.
+   */
+  Map<String, String> processJsonInputs(List<CompilerInput> inputsToProcess) {
+    RewriteJsonToModule rewriteJson = new RewriteJsonToModule(this);
+    for (CompilerInput input : inputsToProcess) {
+      if (!input.getSourceFile().getOriginalPath().endsWith(".json")) {
+        continue;
+      }
+
+      input.setCompiler(this);
+      try {
+        // JSON objects need wrapped in parens to parse properly
+        input.getSourceFile().setCode("(" + input.getSourceFile().getCode() + ")");
+      } catch (Exception e) {
+        continue;
+      }
+
+      Node root = input.getAstRoot(this);
+      if (root == null) {
+        continue;
```

Comment:
My thought was for invalid JSON. Since we wrap the JSON in parens, I think it will always have at least a root node.